internal/health/condition/check_ready.go

```diff
@@ -27,7 +27,7 @@ func (r *readyCheck) Check(_ context.Context, etcd druidv1alpha1.Etcd) Result {
 	}
 
 	var (
-		size         = len(etcd.Status.Members)
+		size         = int(etcd.Spec.Replicas)
```

Comment:
There are two ways to argue this:
1. The status must reflect what the spec demands. If the spec says replicas=3, then the status should reflect whether the etcd cluster is ready in relation to the current spec.
2. Because we allow druid to reconcile the spec upon addition of a reconcile-annotation, it's possible that the spec will not be reconciled for a long time, and during this entire period, the status will show ready condition as false because the cluster has not been scaled up yet. 

Ideally, we need a way to check the current actual etcd cluster size (number of members registered in the cluster). @unmarshall if you remember, we had this discussion recently about bringing back the status.etcdClusterSize field (maybe with a different name), but with the purpose of storing the total number of members that have joined the cluster at any given point of time. I think this becomes more and more relevant now.

But for this PR, let's not include that. status.members is updated with a new member every time a new member joins the cluster and updates its member lease. If the member lease is created but not yet renewed by the backup sidecar responsible for it, then status.members does not reflect that. So I think we can and should continue using `len(etcd.Status.Members)` here, as @ishan16696 pointed out.

Reply:
reverted the change